### PR TITLE
Basic listing of courses in an organization

### PR DESCRIPTION
**Pull request by marcospri**
This allows to have another "extra level" for navigation.

See: https://hypothes-is.slack.com/archives/C4K6M7P5E/p1716983621819659

The approach here is the simplest we can merge with hopefully the goal of unblocking FE work.

## Testing

(Adjust public id accordingly to localhost data, I have a note about making this part of devdata in my pile)

- Entrypoint http://localhost:8001/dashboard/organizations/LDtcl7EUTeW2UERPJLAVtA
- API route: http://localhost:8001/api/dashboard/organizations/LDtcl7EUTeW2UERPJLAVtA/courses

### CODE REVIEW

File: lms/views/dashboard/api/course.py [hunk 1]
```diff
@@ -7,14 +7,29 @@
...
+        request_method="GET",
+        renderer="json",
+        permission=Permissions.DASHBOARD_VIEW,
+    )
+    def get_organization_courses(self) -> list[APICourse]:
+        org = get_request_organization(self.request, self.organization_service)
+        courses = self.course_service.search(
+            limit=None, organization_ids=[org.id], user=self.request.user
+        )
+        return [APICourse(id=course.id, title=course.lms_name) for course in courses]
```
Comment (marcospri) [inline lms/views/dashboard/api/course.py:32]: Just APICourse here.

- No other stats for the course
- No parent org exposed, we don't want to start explicitly exposing the organization level just yet.

File: lms/models/application_instance.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ class ApplicationSettings(JSONSettings):
         ),
         JSONSetting("hypothesis", "instructor_email_digests_enabled", asbool),
         JSONSetting("hypothesis", "instructor_dashboard", asbool),
+        JSONSetting("hypothesis", "lti_13_sourcedid_for_grading", asbool),
```
Comment (acelaya) [inline lms/models/application_instance.py:58]: Is this related to this PR? 🤔

File: lms/routes.py [hunk 3]
```diff
@@ -250,6 +250,11 @@ def includeme(config):  # noqa: PLR0915
         factory="lms.resources.dashboard.DashboardResource",
     )
 
+    config.add_route(
+        "api.dashboard.organizations.courses",
+        "/api/dashboard/organizations/{public_id}/courses",
+    )
```
Comment (acelaya) [inline lms/routes.py:256]: I guess we also need a non-API route to be registered, that returns the "blank" template for the FE app to kick-in

File: lms/services/grading_info.py [hunk 4]
```diff
@@ -49,10 +49,20 @@ def get_students_for_grading(
...
+                if application_instance.settings.get(
+                    "hypothesis", "lti_13_sourcedid_for_grading", False
+                ):
+                    # While all the above is true the situation gets further complicated by the  lti1p1 claim that provides
+                    # the old values of some IDs in LTI1.3
+                    # In this scenario, for students that have launched after the upgrade will get:
+                    #  grading_info.user_id to be the old  LTI user ID, the LTI1.1
+                    #  grading_info.lis_result_sourced_id will be the "sub" parameter via LTIParams
+                    # In that case grading_info.lis_result_sourced_id is the right value to use to talk to the LTI1.3 grading API
+                    lis_result_sourced_id = grading_info.lis_result_sourcedid
```
Comment (acelaya) [inline lms/services/grading_info.py:65]: This looks like it belongs to another PR

File: lms/routes.py [hunk 5]
```diff
@@ -250,6 +250,11 @@ def includeme(config):  # noqa: PLR0915
         factory="lms.resources.dashboard.DashboardResource",
     )
 
+    config.add_route(
+        "api.dashboard.organizations.courses",
+        "/api/dashboard/organizations/{public_id}/courses",
```
Comment (acelaya) [inline lms/routes.py:255]: Can we call the param `organization_id` instead of `public_id`? It will make it more clear when used out of the context of the URL itself.

**Review by acelaya — COMMENTED** [review]
I noticed a couple of things:

* We need to expose a `/dashboard/organizations/{organization_id}` route that serves the dashboard template.
* There are some changes in this PR that apparently belong to another PR. Maybe the base branch should not be `main` or this PR is not up to date with `main`?

File: lms/routes.py (hunk 3 above)
Comment (marcospri) [reply lms/routes.py:256]: Yup :+1:

File: lms/routes.py (hunk 5 above)
Comment (marcospri) [reply lms/routes.py:255]: Went for `organization_public_id` to distinguish  it from the PK.

Added a note to the board about using this naming in all routes.

File: lms/resources/_js_config/__init__.py [hunk 6]
```diff
@@ -266,6 +266,9 @@ def enable_dashboard_mode(self):
                         course_assignment_stats=self._to_frontend_template(
                             "api.dashboard.course.assignments.stats"
                         ),
+                        organization_courses=self._to_frontend_template(
```
Comment (marcospri) [inline lms/resources/_js_config/__init__.py:269]: No _stats here yet so no need for that in the name. Should I added it here to avoid extra work in the future?

File: lms/security.py [hunk 7]
```diff
@@ -374,6 +374,9 @@ def get_lti_user(request) -> LTIUser | None:
 
 
 def _get_user(request) -> User | None:
+    if not request.lti_user:
```
Comment (marcospri) [inline lms/security.py:377]: Allows request.user to return None instead of failing for non LTI users.

File: lms/resources/_js_config/__init__.py (hunk 6 above)
Comment (acelaya) [reply lms/resources/_js_config/__init__.py:269]: Perhaps we should not have added `_stats` to any of them, to leave more room to decide the exact information returned for each one of them.

Let's leave it like this for now, and we can see in future if we need to make adjustments.

File: lms/services/organization.py [hunk 8]
```diff
@@ -383,6 +384,24 @@ def get_hierarchy_ids(self, id_, include_parents=False) -> list[int]:
...
+            .join(Course)
+            .join(ApplicationInstance)
+            .where(
+                GroupingMembership.user_id == user.id,
+                ApplicationInstance.organization_id == organization.id,
+            )
+            .limit(1)
+        )
+        return bool(self._db_session.execute(query).scalar())
+
```
Comment (acelaya) [inline lms/services/organization.py:404]: Perhaps a micro-optimization, but if we just want to know if there's a result for that query, maybe we can do a `SELECT id ...`, or something in those lines, so that the whole model doesn't need to be mapped.

Although, maybe `execute(query).scalar()` does precisely that, so don't give me too much credit 😅 

Or, if you think it's just not worth the effort, I'm also ok with that.

**Review by acelaya — APPROVED** [review]
LGTM 👍🏼

File: lms/services/organization.py (hunk 8 above)
Comment (marcospri) [reply lms/services/organization.py:404]: Yeah, that's a good point although in the grand scheme of things, how the LMS app works, I don't think is worth the effort.
